core/unit_test/TestExecutionEnvironmentNonInitializedOrFinalized.cpp

```diff
@@ -41,21 +41,96 @@ TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest,
     Kokkos::View<int> v0;
     Kokkos::View<float*> v1;
     Kokkos::View<NonTrivial**> v2;
+    return std::make_tuple(v0, v1, v2);
   };
   EXPECT_EXIT(
       {
-        make_views();
+        { (void)make_views(); }
         std::exit(EXIT_SUCCESS);
       },
       ::testing::ExitedWithCode(EXIT_SUCCESS), "");
   EXPECT_EXIT(
       {
-        Kokkos::initialize();
-        Kokkos::finalize();
-        make_views();
+        {
+          Kokkos::initialize();
+          auto views =
+              make_views();  // views outlive the Kokkos execution environment
+          Kokkos::finalize();
+        }
+        std::exit(EXIT_SUCCESS);
+      },
+      ::testing::ExitedWithCode(EXIT_SUCCESS), "");
+  EXPECT_EXIT(
+      {
+        {
+          Kokkos::initialize();
+          Kokkos::finalize();
+          (void)make_views();
+        }
         std::exit(EXIT_SUCCESS);
       },
       ::testing::ExitedWithCode(EXIT_SUCCESS), "");
 }
 
+TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest, views) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  EXPECT_EXIT(
+      {
+        {
+          Kokkos::View<int*> v;
+          Kokkos::initialize();
+          v = Kokkos::View<int*>("v", 10);
+          v = Kokkos::View<int*>();
+          Kokkos::finalize();
+        }
+        std::exit(EXIT_SUCCESS);
+      },
+      ::testing::ExitedWithCode(EXIT_SUCCESS), "");
+  EXPECT_EXIT(
+      {
+        {
+          Kokkos::initialize();
+          Kokkos::View<int*> v("v", 10);
+          v = {};  // assign default constructed view
+          Kokkos::finalize();
+        }
+        std::exit(EXIT_SUCCESS);
+      },
+      ::testing::ExitedWithCode(EXIT_SUCCESS), "");
+  EXPECT_DEATH(
+      {
+        Kokkos::initialize();
+        Kokkos::View<int*> v("v", 0);
+        Kokkos::finalize();
+      },
+      "Kokkos allocation \"v\" is being deallocated after Kokkos::finalize was "
+      "called");
+#ifndef KOKKOS_ENABLE_SYCL
+  std::string matcher2 =
+      "Constructing View and initializing data with uninitialized execution "
+      "space";
+#endif
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP) || \
+    defined(KOKKOS_ENABLE_SYCL) || defined(KOKKOS_ENABLE_OPENACC)
+  std::string matcher1 = std::string("Kokkos::") +
+                         Kokkos::DefaultExecutionSpace::name() +
+                         "::" + Kokkos::DefaultExecutionSpace::name() +
+                         " instance constructor : ERROR device not initialized";
+#else
+  std::string matcher1 = matcher2;
+#endif
+#ifdef KOKKOS_ENABLE_SYCL
+  std::string matcher2 = matcher1;
+#endif
```

Comment:
I find that hard to read and would prefer
```suggestion
  [[maybe_unused ]] auto error_execution_space_not_initialized = "Constructing View and initializing data with uninitialized execution ";
      "space";
 
 #if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP) || \
     defined(KOKKOS_ENABLE_SYCL) || defined(KOKKOS_ENABLE_OPENACC)
  std::string matcher1 = std::string("Kokkos::") +
                         Kokkos::DefaultExecutionSpace::name() +
                         "::" + Kokkos::DefaultExecutionSpace::name() +
                         " instance constructor : ERROR device not initialized";
#else
  std::string matcher1 = error_execution_space_not_initialized;
#endif

#ifdef KOKKOS_ENABLE_SYCL
  std::string matcher2 = matcher1;
#else
  std::string matcher2 = error_execution_space_not_initialized;
#endif
```

Reply:
I don't think that's much better but let's see what others think

Reply:
I find Daniel's easier to read. 

What is the reason for the double `Kokkos::DefaultExecutionSpace::name()::Kokkos::DefaultExecutionSpace::name()` in matcher1?